tests/test.cpp

```diff
@@ -8,12 +8,30 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include "buffer.h"
+#include "task.h"
+
 #include <boost/ut.hpp>
 
+#include <cstring>
+
 using namespace boost::ut;
 
+namespace tn = tasknow;
+
 // NOLINTNEXTLINE
 int main()
 {
+    "task_[un]serializer"_test = [] {
+        const char* str{"abc"};
+        tn::Task task{str};
+
+        tn::Buffer serialized_task = tn::serialize(&task);
+        tn::Task unserialized_task = tn::unserialize(&serialized_task);
+
+        expect(serialized_task.size == 5_i16);
```

Comment:
you should compare `serialized_task.size` to `2 + str.size()`. there is no reason _not_ to use `std::string` for `str` here.